### DISCUSSION TITLE
Inject values from publicRuntimeConfig into meta

### DISCUSSION THREAD

**Discussion by Xowap** [nuxt/nuxt#18022] [Questions]
Hi,

I'm trying to have something like this in my `nuxt.config.js` (using Nuxt 3):

```
    meta: {
        script: [
            {
                src: "https://plausible.io/js/plausible.js",
                defer: "",
                "data-domain": INSERT_DOMAIN_HERE,
            },
        ],
    },
```

The problematic part is `INSERT_DOMAIN_HERE`. I'm not sure how to read it from the `publicRuntimeConfig`.

I've got a few ideas on how to solve this problem, namely:

- Try to find a hook, I've seen a [`meta:register` hook](https://github.com/nuxt/framework/blob/main/packages/nuxt3/src/app/nuxt.ts#L18), however it doesn't seem to get called
- Use a plugin and somehow hijack the meta config, although I'm not sure how safe it can be
- Override the HTML template, I've done this in Nuxt 2 and it works but for this case that sounds fairly excessive

I'm curious what would be the _right_ way to do this?

Thanks

**Comment by kiranimeid** [comment]
why not using .env? and call it from there using process.env?

**Reply by Xowap** [reply]
For the same reasons that make `publicRuntimeConfig` exist, however I'm not sure if it's relevant for `meta` now that you'll say it. I'll give it a try thanks.

**Reply by Xowap** [reply]
Well I've tried it out, it didn't work. However, I've also noticed that `publicRuntimeConfig` equally isn't evaluated at build time, which is definitely unsettling

**Comment by maxdzin** [comment]
@Xowap Here's an example of how I used `publicRuntimeConfig` for the meta:
```
<script setup lang="ts">
  const config = useRuntimeConfig()

  useMeta({
    title: config.APP_NAME,
    ...
  })
</script>
```

and in the `nuxt.config.ts`:
```
export default defineNuxtConfig({
  publicRuntimeConfig: {
    APP_NAME: 'Example app',
    ...
  },
}
```
